types/http-proxy-agent/index.d.ts

```diff
@@ -10,6 +11,11 @@ import { Url } from 'url';
 
 declare class HttpProxyAgent extends Agent {
     constructor(options: string | Url);
+
+	options: Url;
```

Comment:
I don't see these properties in agent implementation: `options`, `timeout`. This is the hidden/private field inside the agent (`http.Agent`)? If so, these types must be described in the [`node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node) package.

Reply:
Thanks for your review.

I am not sure where these properties come from. I can see that the `http-proxy-agent` declares `var proxy` and that is the only one I need. I'm happy to remove the other properties if this is what you were suggesting.

The way I found these properties was by logging the agent.

```typescript
let agent = options.agent as HttpProxyAgent
console.log(agent)
```

That give this output

```typescript
HttpProxyAgent {
  _events: {},
  _eventsCount: 0,
  _maxListeners: undefined,
  _promisifiedCallback: true,
  timeout: null,
  options:
   Url {
     protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost:30002',
     port: '30002',
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     pathname: '/',
     path: '/',
     href: 'http://localhost:30002/' },
  secureProxy: false,
  proxy:
   { protocol: 'http:',
     slashes: true,
     auth: null,
     host: 'localhost',
     port: 30002,
     hostname: 'localhost',
     hash: null,
     search: null,
     query: null,
     href: 'http://localhost:30002/' },
  callback: [Function] }
```

Reply:
This is should be a part of the Node.js typings:

https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L51